server.py

```diff
@@ -55,169 +52,142 @@ def get(self, feature_id):
-      template_data['FEATUREOBSERVER_BUCKETS'] = json.dumps(
-          properties, separators=(',',':'))
-    elif path.startswith('omaha_data'):
-      omaha_data = util.get_omaha_data()
-      return common.JSONHandler.get(self, omaha_data, formatted=True)
-
-    if path.startswith('metrics/'):
-      push_urls = http2push.use_push_manifest('push_manifest_metrics.json')
-
-    # Add Link rel=preload header for h2 push on .html file requests.
-    if push_urls:
-      self.response.headers.add_header(
-          'Link', self._generate_link_preload_headers(push_urls))
-
-    self.render(data=template_data, template_path=os.path.join(path + '.html'))
-
-
-class FeaturesAPIHandler(common.JSONHandler):
-
-  def get(self, version=None):
-    ramcache.check_for_distributed_invalidation()
-    if version is None:
-      version = 2
-    else:
-      version = int(version)
+    template_data['categories'] = [
+      (v, normalized_name(v)) for k,v in
+      models.FEATURE_CATEGORIES.iteritems()]
+    template_data['IMPLEMENTATION_STATUSES'] = json.dumps([
+      {'key': k, 'val': v} for k,v in
+      models.IMPLEMENTATION_STATUS.iteritems()])
+    template_data['VENDOR_VIEWS'] = json.dumps([
+      {'key': k, 'val': v} for k,v in
+      models.VENDOR_VIEWS.iteritems()])
+    template_data['WEB_DEV_VIEWS'] = json.dumps([
+      {'key': k, 'val': v} for k,v in
+      models.WEB_DEV_VIEWS.iteritems()])
+    template_data['STANDARDS_VALS'] = json.dumps([
+      {'key': k, 'val': v} for k,v in
+      models.STANDARDIZATION.iteritems()])
+
+    return template_data
+
+
+
+class CssPopularityHandler(common.FlaskHandler):
+
+  TEMPLATE_PATH = 'metrics/css/timeline/popularity.html'
+
+  def get_template_data(self, bucket_id=None):
+    # Note: bucket_id is not used, but the JS looks in the URL to get it.
+    properties = sorted(
+        models.CssPropertyHistogram.get_all().iteritems(), key=lambda x:x[1])
+    template_data = {
+        'CSS_PROPERTY_BUCKETS': json.dumps(
+            properties, separators=(',',':')),
+        }
+    return template_data
+
+
+class CssAnimatedHandler(CssPopularityHandler):
+
+  TEMPLATE_PATH = 'metrics/css/timeline/animated.html'
+  # The logic and data is the same, but it is filtered differenly in JS.
 
+
+class FeaturePopularityHandler(common.FlaskHandler):
+
+  TEMPLATE_PATH = 'metrics/feature/timeline/popularity.html'
+
+  def get_template_data(self, bucket_id=None):
+    # Note: bucket_id is not used, but the JS looks in the URL to get it.
+    properties = sorted(
+        models.FeatureObserverHistogram.get_all().iteritems(), key=lambda x:x[1])
+    template_data = {
+        'FEATUREOBSERVER_BUCKETS': json.dumps(
+            properties, separators=(',',':')),
+    }
+    return template_data
+
+
+class OmahaDataHandler(common.FlaskHandler):
+
+  JSONIFY = True
+
+  def get_template_data(self):
+    omaha_data = util.get_omaha_data()
```

Comment:
Does this hit memcache?

Reply:
Yes, it is cached.  I removed memcache and we now use ramcache.py.